### DISCUSSION TITLE
Does Lem-SDL allow for a transparent background?

### DISCUSSION THREAD

**Discussion by aykaramba** [lem-project/lem#748] [General]
This is a triviality and a non issue, no dev should spend any time thinking about this please.  Just curious:

Does Lem-SDL allow for a transparent colour for the background so that I can see windows/wallpaper behind Lem?

I run StumpWM + URXVT and I have setup URXVT to have a transparent background.  Because I am running StumpWM, URXVT always renders the wallpaper as it is a tiling window manager and there is no such thing as an "overlapping" window so I always get a nice random background in my emacs-nox depending on what wallpaper is rotated in and what the placement of emacs-nox is.  I am not sure this would be useful for floating windows because the text of windows below Lem would cause visual confusion. 

Wallpaper support in Lem wouldn't be what I am looking because a transparent URXVT always shows me a different portion of the wallpaper depending on placement which results in nicely unpredictable "views" with code sort of floating on top in a lovely way. 

I tried #26262680 for a 50% css style transparent colour but Lem crashes on that.

Any suggestions are welcome, even a solid  no is perfectly fine.

**Comment by cxxxr** [comment]
I checked the sdl2 API and it seems that a function is provided to make the window transparent.
https://wiki.libsdl.org/SDL2/SDL_SetWindowOpacity

I don't know if this will meet your expectations, but it can be transparent as follows
```
(sdl2-ffi.functions:sdl-set-window-opacity (lem-sdl2::display-window lem-sdl2::*display*) 0.9)
```

**Comment by vindarel** [comment]
> Does Lem-SDL

Hey, you get a transparent background with Lem in the terminal right? 

> but it can be transparent as follows

OMG, it works^^

**Reply by aykaramba** [reply]
Oh wow.  

I am running a Lem binary, can that be evaled from withing the Lem environment?

**Reply by cxxxr** [reply]
You can start the REPL with `M-x start-lisp-repl` to evaluate it.
Or how about writting it in $HOME/.lem/init.lisp?

**Reply by aykaramba** [reply]
Whoa.  That looks amazing.  Note to anyone using StumpWM, only works in XFCE and not StumpWM for me.  Will dig into that some more later, this is kind of changes the game in a bunch of ways.  

Thanks for that!

**Reply by adamgender0** [reply]
Good'
